packages/vite-environment-provider-cloudflare/src/index.ts

```diff
@@ -224,41 +235,33 @@ async function createCloudflareDevEnvironment(
       fixedSpecifier = rawSpecifier;
 
       try {
-        let { id } = await devEnv.pluginContainer.resolveId(
-          fixedSpecifier,
-          referrer,
-          {
-            // The following is to let know `resolveId` if the import is actually a require
-            // https://github.com/vitejs/vite/blob/8851d9d1c97cdce0807edd45e33e70446e545956/packages/vite/src/node/plugins/resolve.ts#L228-L230
-            // (Note: I am not sure if this is actually making any difference ¯\_(ツ)_/¯)
-            custom: {
-              'node-resolve': { isRequire: resolveMethod === 'require' },
-            },
-          },
-        );
-
-        const resolvedId = id;
+        const resolveId =
+          resolveMethod === 'import' ? esmResolveId : cjsResolveId;
+        let resolvedId = await resolveId(devEnv, fixedSpecifier, referrer);
 
-        if (id.includes('?')) id = id.slice(0, id.lastIndexOf('?'));
+        if (resolvedId.includes('?'))
+          resolvedId = resolvedId.slice(0, resolvedId.lastIndexOf('?'));
 
         const redirectTo =
-          id !== rawSpecifier && id !== specifier ? id : undefined;
+          resolvedId !== rawSpecifier && resolvedId !== specifier
+            ? resolvedId
+            : undefined;
 
         if (redirectTo) {
           return new MiniflareResponse(null, {
-            headers: { location: id },
+            headers: { location: resolvedId },
             status: 301,
           });
         }
 
         // and we read the code from the resolved file
-        const code: string | null = await readFile(id, 'utf8').catch(
+        const code: string | null = await readFile(resolvedId, 'utf8').catch(
```

Comment:
Should we surface an error here? We can look at error handling in a separate PR if you prefer.

Reply:
since we didn't have error handling before as well if you'd be ok with it I'd leave it for a follow up PR 🙂